tests/python_functional/src/syslog_ng_config/statements/destinations/file_destination.py

```diff
@@ -25,19 +25,21 @@
 import src.testcase_parameters.testcase_parameters as tc_parameters
 from src.driver_io.file.file_io import FileIO
 from src.syslog_ng_config.statements.destinations.destination_driver import DestinationDriver
+from src.syslog_ng_config.statements.destinations.destination_reader import DestinationReader
 
 
 class FileDestination(DestinationDriver):
     def __init__(self, file_name, **options):
         self.driver_name = "file"
         self.path = Path(tc_parameters.WORKING_DIR, file_name)
-        super(FileDestination, self).__init__(FileIO, [self.path], options)
+        super(FileDestination, self).__init__([self.path], options)
+        self.destination_reader = DestinationReader(FileIO)
```

Comment:
That looks nice.

Reply:
The `DestinationDriver` class just became almost empty after extracting the reader from it. Do you have plans to add new functionality to `DestinationDriver`?

Reply:
Yes, you are right. In my next PR I have a plan to extend SourceDriver() and DestinationDriver() classes with their new scopes. 

I really do not know if that change can fit here (it could extend this PR a lot).
"Unfortunately" this classes FileSource / FileDestination / SourceDriver / DestinationDriver / SourceWriter / DestinationReader are highly depends on each other.

One other note, I closed my previous PR because of that "huge" change.

I will think over it again. What is your suggestion after this?

Reply:
Nothing, I was just curious. This class seems to have its place, I'm sure it'll have the right functionality later :)